helm/hpcc/templates/_helpers.tpl

```diff
@@ -1045,11 +1045,42 @@ Pass in dict with root, category.  optional name to restrict it to a single name
 
 {{/*
 Create placement related settings
-Pass in dict with placement
+Pass in dict with current placment and new placements dictinary for this entity
+*/}}
+{{- define "hpcc.addPlacementSetting" -}}
+{{- if .me.placement.nodeSelector }}
+{{- $_ := set .new "nodeSelector" (merge (.new.nodeSelector | default dict ) .me.placement.nodeSelector)  }}
+{{- end -}}
+{{- if .me.placement.tolerations }}
+{{- $_ := set .new "tolerations" (concat (.new.tolerations | default list ) .me.placement.tolerations)  }}
```

Comment:
should this remove duplicates? (i.e. with | uniq)

Reply:
what happens if there are multiple "pods" matches for tolerations that define the same named key value, meaning that the merged list has multiple entries for the same key(s) ?

Reply:
"if or (has $target $placement.pods) (has $type $placement.pods) (has "all" $placement.pods) -}}" will identify which pod/job has the settings.  So it is OK if the same tolerations/nodeSelector is set for multiple pod. But if the same tolerations/nodeSelector  is applied to the same pod more than once the code currently cannot detect it and it will be appended multiple times.

Reply:
could probably avoid generating duplciates by performing a uniq for tolerations and a deep merge for nodeSelector,
but I think it's ok without doing that, it will mean the generated yaml could have duplicates, but it should still parse/succeed albeit with last duplicate nodeSelector taking effect in that situation.

Reply:
I add this as a limitation in placements.md as mentioned in above.